cpp/src/io/json/read_json.cu

```diff
@@ -41,6 +42,57 @@ namespace cudf::io::json::detail {
 
 namespace {
 
+class compressed_host_buffer_source final : public datasource {
+ public:
+  explicit compressed_host_buffer_source(std::unique_ptr<datasource> src, compression_type comptype)
+    : _comptype{comptype}
+  {
+    auto dbuf  = src->host_read(0, src->size());
+    _ch_buffer = std::vector<std::uint8_t>(dbuf->data(), dbuf->data() + dbuf->size());
```

Comment:
Can we store `_ch_buffer` as `datasource::buffer` directly? Doing so, we can just move `dbuf`.

Reply:
Thank you, the datasource class now directly stores the buffer.